definitions/marts/looker_studio_marts/ecf/ls_ecf_mentor_backfill_grant_funding.sqlx

```diff
@@ -0,0 +1,212 @@
+        
+  LEFT JOIN
+    ${ref('ecf_inductions')} ind2
+  ON
+    ind2.participant_profile_id = dec.participant_profile_id
+    AND
+    DATE(dec.declaration_date) < IFNULL(DATE(ind2.end_date), '2050-12-31')
+    AND
+    ind.induction_record_id IS NULL
+
+  LEFT JOIN
+    ${ref('ls_declarations_provider_names')} AS dp
+  ON
+    dp.id = dec.declaration_id
+
+  WHERE
+    funded_declaration = TRUE
+    AND
+    participant_course = 'Mentor'
+    AND
+    dec.declaration_date >= '2023-09-01'
+  QUALIFY (
+    rn0 = 1
+  )
+),
+
+ects AS (
+  SELECT 
+     sch.urn
+    ,COUNT(1) AS school_ect_count 
+  FROM 
+    ${ref('participant_profiles_latest_cpd')} AS pp
+  LEFT JOIN
+    ${ref('school_cohorts_latest_cpd')} AS scho
+  ON
+    scho.id = pp.school_cohort_id
+  LEFT JOIN
+    ${ref('schools_latest_cpd')} AS sch 
+  ON
+    sch.id = scho.school_id
+  WHERE
+    type LIKE '%ECT'
+    AND
+    induction_completion_date IS NULL OR induction_completion_date > '2023-08-31'
+  GROUP BY
+    sch.urn
+  ORDER BY 
+    school_ect_count DESC
+),
+
+declarations_extended AS (
+  SELECT
+     dec.*
+    ,ROW_NUMBER() OVER(PARTITION BY dec.participant_profile_id ORDER BY declaration_type_hierarchy ASC) AS declaration_sequence
+    ,ROUND((cost.cost_for_36_hours / 36) * dec.funded_hours, 2) AS funded_hours_cost
+    ,CASE
+      WHEN pp.user_id IS NOT NULL THEN TRUE
+      ELSE FALSE
+     END AS previously_paid
+    ,gias.LA_Name AS local_authority
+    ,gias.LA_code AS la_code
+    ,band.teacher_pay_band
+    ,gias.EstablishmentName AS gias_school_name
+    ,gias.TypeOfEstablishment_name AS gias_establishment_type
+    ,gias.PhaseOfEducation_name AS gias_phase_of_education
+    ,gias.EstablishmentStatus_name AS gias_establishment_status
+    ,COUNT(DISTINCT dec.participant_profile_id) OVER(PARTITION BY dec.used_school_urn) AS school_mentor_count
+    ,ects.school_ect_count
+  FROM
+    declarations dec
+  LEFT JOIN
+    ects 
+  ON 
+    ects.urn = dec.used_school_urn
+  LEFT JOIN
+    ${ref('establishments')} AS gias
+  ON
+    CAST(gias.urn AS INT) = dec.used_school_urn
+  LEFT JOIN
+    ${ref('la_teacher_pay_band_grouping')} AS band
+  ON
+    band.LA_code = CAST(gias.LA_code AS INT)
+  LEFT JOIN
+    ${ref('ecf_mentors_previously_paid')} AS pp
+  ON
+    pp.user_id = dec.user_id
+    AND 
+    (
+      (
+        dec.declaration_type IN ('started', 'retained-1', 'retained-2')
+        AND
+        pp.payment_period LIKE 'Instalment 1%'
+      )
+      OR
+      (
+        dec.declaration_type IN ('retained-3', 'retained-4', 'completed')
+        AND
+        pp.payment_period LIKE 'Instalment 2%'
+      )
+    )
+
+  LEFT JOIN
+    ${ref('teacher_pay_band_36_hour_cost')} AS cost
+  ON
+    cost.teacher_pay_band = band.teacher_pay_band
+)
+
+SELECT
+   * EXCEPT(rn0)
+  -- Funded hours per school for mentors not previously paid to enable aggregation at school level for relevant cases only.

```

Comment:
Sorry, bit confused here, is it enabling future aggregation or is it aggregating at this point? It looks to me like it's aggregating it at this point by grouping by the school_urn? So in theory, all declarations for the same school will have the same value for this field?